doc/content/whats-new/3.12.2.md

```diff
@@ -0,0 +1,29 @@
+---
+date: 2021-04-30
+title: "3.12.2"
+---
+
+### Added
+
+- Contextual tooltips to form fields in the Console.
+- C-Style uint32_t representation for end device address field.
+- Gateway Configuration Server to the cluster package.
+  - This introduces a new config option `cluster.gateway-configuration-server` that needs to be set in multi-instance deployments.
+- Uplink storage for integrations in the Application Server. The number of uplinks stored per end device may be configured via the config option `as.uplink-storage.limit`.
+- LoRaCloud GLS multi frame request support.
+- LoRaCloud GNSS request support.
+- LoRaCloud WiFi request support.
+
+### Changed
+
+- Allow the LinkADRReq commands to lower the data rate used by the end devices.
+
+### Fixed
+
+- Occasional crashes in the ratelimit middleware.
+- Handling of zero EUI CUPS update-info requests.
+- Backend validation messages for some forms.
+- Gateway downlink message previews not displaying correctly in the event view of the Console.
+- Importing end devices from the Console would occasionally ignore some device MAC settings fields.
+
+### Security
```

Comment:
Shouldn't we remove empty sections?

Reply:
These aren't removed for previous releases, so I will leave this for now. If we decide to remove those, I can easily remove it in another PR